### PR TITLE
[4.0] Spacer

### PR DESCRIPTION
**Pull request by brianteeman**
PR for #21401

Thanks for the code @infograf768

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on 704fd7ef32d59ad10975effd1a2a168ead3a19c9<br /><br />Same Result on /administrator/index.php?option=com_config&view=component&component=com_media as without PR.

screen shot 2018-08-06 at 20 23 05


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21425">issues.joomla.org/tracker/joomla-cms/21425</a>.</sub>

**Comment by brianteeman** [comment]
closed while i fix the build

**Comment by brianteeman** [comment]
It looks correct so I have no idea why its not working - hopefully someone can see the error

File: administrator/templates/atum/scss/pages/_com_config.scss [hunk 1]
```diff
@@ -12,4 +12,10 @@
       margin-bottom: 2rem;
     }
   }
+
+  .field_spacer {
```
Comment (ggppdk) [inline administrator/templates/atum/scss/pages/_com_config.scss]: dash instead of underscore and it should work
`.field-spacer`

Comment (ggppdk) [reply administrator/templates/atum/scss/pages/_com_config.scss]: But why fix this only for com_config ?
I see same texts via spacer in XML files of various menu items

**Comment by brianteeman** [comment]
updated now

**Comment by ggppdk** [comment]
It works, but same need in menu items

what about fixing this for all screens by adding to
template.scss
template-rtl.scss

or to other appropriate scss file ?
```css
.control-group {
	&.field-spacer {
		.control-label {
			width: auto;
		}
	}
}
```

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 3954198acbcae3a0e482cf798013e9808cbe834d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21425">issues.joomla.org/tracker/joomla-cms/21425</a>.</sub>

**Comment by infograf768** [comment]
Agree with @ggppdk
The fix should be done as a general fix not as a specific com_config one.

For the moment the issue is only present in 2 places, but it may be necessary elsewhere too in the future.

**Comment by ggppdk** [comment]
hhmm in more than 2 places ...

Searching for (type="spacer" **that also has label**) 

```
Search "type="spacer"\s+label="" (24 hits in 17 files)

administrator\components\com_categories\forms\category.xml (1 hit)
	Line 126: 		type="spacer"

administrator\components\com_fields\forms\field.xml (1 hit)
	Line 102: 			type="spacer"

administrator\components\com_media\config.xml (1 hit)
	Line 27: 			type="spacer" 

components\com_contact\forms\contact.xml (1 hit)
	Line 6: 			type="spacer"

components\com_contact\tmpl\categories\default.xml (2 hits)
	Line 97: 				type="spacer" 
	Line 187: 				type="spacer" 

components\com_contact\tmpl\category\default.xml (2 hits)
	Line 40: 				type="spacer" 
	Line 132: 				type="spacer" 

components\com_contact\tmpl\featured\default.xml (1 hit)
	Line 22: 				type="spacer" 

components\com_content\tmpl\categories\default.xml (3 hits)
	Line 101: 				type="spacer" 
	Line 201: 				type="spacer"
	Line 315: 				type="spacer" 

components\com_content\tmpl\category\blog.xml (1 hit)
	Line 173: 					type="spacer"

components\com_content\tmpl\featured\default.xml (1 hit)
	Line 35: 				type="spacer" 

components\com_newsfeeds\tmpl\categories\default.xml (2 hits)
	Line 100: 				type="spacer" 
	Line 192: 				type="spacer" 

components\com_newsfeeds\tmpl\category\default.xml (2 hits)
	Line 38: 			type="spacer" 
	Line 128: 				type="spacer" 

components\com_search\tmpl\search\default.xml (1 hit)
	Line 64: 				type="spacer"

components\com_tags\tmpl\tag\default.xml (1 hit)
	Line 130: 				type="spacer" 

components\com_tags\tmpl\tag\list.xml (1 hit)
	Line 129: 				type="spacer" 

components\com_users\forms\registration.xml (1 hit)
	Line 6: 			type="spacer"

plugins\user\profile\profile.xml (2 hits)
	Line 26: 					type="spacer"
	Line 169: 					type="spacer"
```

**Comment by brianteeman** [comment]
@ggppdk My quick check of a sample of the items you link to is that in all cases they are using the spacer for a title as intended and not for a long description which is what this pr addresses

**Comment by ggppdk** [comment]
@brianteeman 

>My quick check of a sample of the items you link to is that in all cases they are using the spacer for a title as intended and not for a long description which is what this pr addresses

i understand that for majority of case it is as you say, but not for all cases
e.g. create a menu item "Featured contents",
and go to 2nd TAB "List layouts", it is description

Is there any harm / reason not to fix this for all cases ?
adding to appropriate scss file ?

```css
.control-group {
	&.field-spacer {
		.control-label {
			width: auto;
		}
	}
}
```

**Comment by brianteeman** [comment]
dont have time to debate that the spacer field is being abused - feel free to submit your own pr

**Comment by ghost** [comment]
Issue #21401 reopened.

**Pull request closed without merging**